scripts/update_all_products.pl

```diff
@@ -1201,10 +1209,11 @@
 				}
 
 				# Store data to mongodb
-				# Make sure product code is saved as string and not a number
+				# Make sure product _id and code are saved as string and not a number
 				# see bug #1077 - https://github.com/openfoodfacts/openfoodfacts-server/issues/1077
 				# make sure that code is saved as a string, otherwise mongodb saves it as number, and leading 0s are removed
-				$product_ref->{code} = $product_ref->{code} . '';
+				$product_ref->{_id} .= '';
+				$product_ref->{code} .= '';
 				$products_collection->replace_one({"_id" => $product_ref->{_id}}, $product_ref, { upsert => 1 });
```

Comment:
Should'nt we refactor and have an upsert_product function that ensure this kind of thing ? (Instead of using replace_one directly). This may help avoiding this error and maybe others ?

Reply:
Good idea. It could be a good issue for open source day. cc @yuktea 
I filed a bug for it: https://github.com/openfoodfacts/openfoodfacts-server/issues/7271